v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -345,6 +365,24 @@ func init() {
 				return strings.Join(stringElements, separator), nil
 			},
 		),
+		"join_slice": makeDslWithOptionalArgsFunction(
```

Comment:
> However, I cannot use `join()` because it accepts a variable number of string parameters, whereas `sort()` returns 1 slice of string. Instead, I created another DSL `join_slice()` and did `join_slice(" ", sort("b", "a", "2", "c", "3", "1", "d", "4"))`

Couldn't the same logic be integrated into the existing `join` function, instead of creating a new one?

Reply:
Done.

New `join()` behavior:
- if < 2 args are given, return error
- if 2 args are given, join as slice of string
- if > 2 args are given, join as strings/ints/etc.
- if exception occurs at any point (such as unexpected type), return error.